examples/advanced/maximal_propagate.py

```diff
@@ -0,0 +1,39 @@
+"""
+Example showing repeated solving for deriving the maximal consequence of set of constraints.
+
+Iteratively finds a new solutions by forcing at least one variable to have a value not seen in any other solution.
+"""
+
+from cpmpy import *
+from cpmpy.transformations.get_variables import get_variables
+
+
+def maximal_propagate(constraints, solvername="ortools"):
+    # For faster propagation, use incremental solver such as pysat, z3 or gurobi.
+    vars = get_variables(constraints)
+    visisted_domain = {var: set() for var in vars}
+
+    solver = SolverLookup.get(solvername)
+    solver += constraints
+
+    while solver.solve():
+        cons = False
```

Comment:
Ideally, we should be able to solve with an assumption variable that allows for temporarily turning on/off the added constraints:

`solver.solve(assumptions=[bv])`